### PR TITLE
fix(exceptions): Fixes #1667 add .catch to promises

### LINKED ISSUE #1667: Add .catch to promises for better exception handling
Go in and add .catch to our promises, where we can then do as simple as a Cu.reportError, so we can at least see the exceptions that are thrown

### PR DESCRIPTION
**Pull request by sarracini**
Adding some catch enforcement
So this patch only enforces the catching of promises - we should file another PR for logging it to a central place, once we decide what that is, so that we can essentially mimic what sentry does.

@k88hudson r?

### CODE REVIEW

File: addon/MetadataParser.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ MetadataParser.prototype = {
     return new Promise(resolve => {
       const doc = this._getDocumentObject(raw);
       resolve(this._formatData(getMetadata(doc, url), url));
-    });
+    }).catch(err => Cu.reportError(err));
```
Comment (sarracini) [inline addon/MetadataParser.js]: I think it's ok, both ```getMetadata``` and ```_formatData``` always return something, and if either of them throw an exception we'll be able to catch it here. So I *think* it's fine....

**Comment by k88hudson** [comment]
Overall looks good, I will take a closer look as soon as I finish up some tests (I just want to make sure we aren't prematurely catching anything that needs a specific catch handler)

File: addon/MetadataParser.js (hunk 1 above)
Comment (k88hudson) [inline addon/MetadataParser.js]: If you handle the error here, the error handling in `_asyncParseAndSave` in `PageScraper.js` will never get hit

File: addon/PreviewProvider.js [hunk 2]
```diff
@@ -192,7 +192,7 @@ PreviewProvider.prototype = {
     }
 
     return this._asyncGetEnhancedLinks(processedLinks, event).then(
-      cachedLinks => this._getFaviconColors(cachedLinks));
+      cachedLinks => this._getFaviconColors(cachedLinks)).catch(err => Cu.reportError(err));
```
Comment (k88hudson) [inline addon/PreviewProvider.js]: same thing here with the `catch` never getting hit

Comment (pdehaan) [inline addon/PreviewProvider.js]: I think if you `.catch()` a promise error, it should then resume onto the next chained `.then()`, IIRC.

So in places where we return a promise, not sure if we'd want a catch + log + throw action, so the usage side can handle any errors as well.

**Pull request merged**